### DISCUSSION TITLE
ModelSerializer: Using a class that does not inherit from ChoiceField strips allow_blank

### DISCUSSION THREAD

**Discussion by amhn** [encode/django-rest-framework#8997] [Potential Issue]
While trying to improve schema generation for netbox, I stumbled on something:
I wanted to change the default serializer for fields with choices, so in a subclass of ModelSerializer I set serializer_choice_field to `netbox.api.fields.ChoiceField` which is _not_ a subclass of `rest_framework.fields.ChoiceField`.

Everything looked fine except for fields with `allow_blank=True`. This was omitted from the schema.

I think this is because of the following check:

https://github.com/encode/django-rest-framework/blob/e2a4559c03247f69e85b2e0e7ede4e8b6201adc2/rest_framework/serializers.py#L1296-L1298

I think either this check should be changed to use `self.serializer_choice_field` instead of `ChoiceField` or  moved before Line  1272:
https://github.com/encode/django-rest-framework/blob/e2a4559c03247f69e85b2e0e7ede4e8b6201adc2/rest_framework/serializers.py#L1272-L1275

Did I miss something?

**Comment by auvipy** [comment]
what issue is it causing actually?

**Reply by amhn** [reply]
> Everything looked fine except for fields with allow_blank=True. This was omitted from the schema.

And (I assume) the field does not accept blank values afterwards (not tested).

**Reply by auvipy** [reply]
can you start with a failing test?

**Reply by amhn** [reply]
Sorry. The result was not, what I had in mind and the changes are already reverted. Just thought to give you a heads up for the bug.

With the original ChoiceField I ended up with `ChoiceField(allow_blank=True, [...])` as serializer for a CharField with allow_blank=True and choices set, with the ChoiceField from netbox it was `ChoiceField([...])` for the same model field.
